### PR TITLE
block check current epoch

### PR DESCRIPTION
**Pull request by eesmerdag**

### CODE REVIEW

File: pipeline/payload.go [hunk 1]
```diff
@@ -34,6 +34,7 @@ type HeightPayload struct {
 	DelegationsByValidator map[string][]near.Delegation
 
 	CurrentEpoch         bool
+	firstBlockOfNewEpoch bool
```
Comment (sosedoff) [inline pipeline/payload.go:37]: Why not make it consistent with other fields and capitalize it?

File: pipeline/task_parser.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package pipeline
 
 import (
 	"context"
+	"errors"
```
Comment (sosedoff) [inline pipeline/task_parser.go:5]: Make sure the imports are split in the order: stdlib, others, ours

File: model/validator_epoch_reward.go [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+	"time"
+
+	"github.com/figment-networks/near-indexer/model/types"
+)
+
+type ValidatorEpochReward struct {
+	ID                int64        `json:"-"`
+	AccountID         string       `json:"-"`
+	Epoch             string       `json:"epoch"`
+	DistributedHeight types.Height `json:"distributed_height"`
```
Comment (sosedoff) [inline model/validator_epoch_reward.go:13]: Did you mean `DistributedAtHeight` and `DistributedAtTime` ? seems like a better name

File: store/migrations/013_create_delegator_epochs.sql [hunk 4]
```diff
@@ -4,8 +4,8 @@ CREATE TABLE delegator_epochs (
   account_id             VARCHAR NOT NULL, -- delegator
   validator_id           VARCHAR NOT NULL,
   epoch                  VARCHAR NOT NULL,
-  last_height            INTEGER NOT NULL,
-  last_time              TIMESTAMP WITH TIME ZONE NOT NULL,
+  distributed_height     INTEGER NOT NULL,
+  distributed_time       TIMESTAMP WITH TIME ZONE NOT NULL,
```
Comment (sosedoff) [inline store/migrations/013_create_delegator_epochs.sql:8]: are we sure they never going to be empty?

File: store/migrations/013_create_delegator_epochs.sql [hunk 5]
```diff
@@ -17,5 +17,7 @@ CREATE INDEX idx_delegator_epochs_account_id
 CREATE UNIQUE INDEX idx_delegator_epochs_account_epoch
   ON delegator_epochs(account_id, validator_id, epoch);
 
+CREATE INDEX idx_delegator_epochs_distributed_time ON delegator_epochs (distributed_time);
```
Comment (sosedoff) [inline store/migrations/013_create_delegator_epochs.sql:20]: follow the same formatting as above statement

File: store/queries/validators_rewards.sql [hunk 6]
```diff
@@ -1,11 +1,11 @@
 SELECT
-	to_char(last_time, $INTERVAL) AS interval,
```
Comment (sosedoff) [inline store/queries/validators_rewards.sql:2]: need to get rid of tabs in this file

File: store/validator_aggs.go [hunk 7]
```diff
@@ -121,6 +121,20 @@ func (s ValidatorAggsStore) ImportValidatorEpochs(records []model.ValidatorEpoch
 			r.Efficiency,
 			r.StakingBalance,
 			r.RewardFee,
+		}
+	})
+}
+
+// ImportValidatorEpochsRewards imports validator epochs rewards records in batch
+func (s ValidatorAggsStore) ImportValidatorEpochsRewards(records []model.ValidatorEpochReward) error {
```
Comment (sosedoff) [inline store/validator_aggs.go:129]: any idea on how many records are we importing in a single insert? if we're talking about 1000s then we we need to implement batching

File: pipeline/payload.go (hunk 1 above)
Comment (eesmerdag) [reply pipeline/payload.go:37]: 👍

File: pipeline/task_parser.go (hunk 2 above)
Comment (eesmerdag) [reply pipeline/task_parser.go:5]: 👍

File: model/validator_epoch_reward.go (hunk 3 above)
Comment (eesmerdag) [reply model/validator_epoch_reward.go:13]: absolutely better English 😄 👍

File: store/migrations/013_create_delegator_epochs.sql (hunk 5 above)
Comment (eesmerdag) [reply store/migrations/013_create_delegator_epochs.sql:20]: 👍

File: pipeline/task_parser.go [hunk 8]
```diff
@@ -69,31 +71,45 @@ func (t ParserTask) Run(ctx context.Context, payload *Payload) error {
...
 				}
-				validator.Reward = res
-				remainingRewards = validator.Stake.Sub(res)
 			}
 
 			parsed.Validators = append(parsed.Validators, *validator)
 
-			if delegations, ok := h.DelegationsByValidator[v.AccountID]; ok && remainingRewards.Int != nil {
+			if delegations, ok := h.DelegationsByValidator[v.AccountID]; ok && remainingRewards.Int != nil && h.FirstBlockOfNewEpoch && h.PreviousBlock != nil {
+				if h.PreviousBlock == nil {
```
Comment (sosedoff) [inline pipeline/task_parser.go:96]: this part will never happen as you're checking for PreviousBlock in the if statement above

Comment (eesmerdag) [reply pipeline/task_parser.go:96]: that is true! i am removing that part :)

**Pull request merged**